packages/types/src/proof/zkProof.ts

```diff
@@ -13,14 +13,19 @@ export interface ZkInputs {
   merkleProofArtifactsOrStorageAddress: string | MerkleProofArtifacts;
 }
 
+// TODO: that is a redandant
 export interface ZKProofPayload {
```

Comment:
Why is it redundant?

Reply:
The intention is to optimize it later and not to have repeated properties inside types in general. As you can see that: 
```ts
  // types related to ZkInputs
  circuitFilePath: string;
  verificationKey: string;
  zkeyFilePath: string;
```

```ts
  // types related to Semaphore
  externalNullifier: string;
  signal: string;
```

```ts
  // types related to ZkMetadata
    urlOrigin: string;
```


```ts
  // types related to ZkInputs
  merkleStorageAddress?: string;
```

So there is a optimization needs to be happen here, and will check that in this PR: #673